pkg/controller/networkpolicy/clusternetworkpolicy.go

```diff
@@ -254,15 +258,154 @@ func (n *NetworkPolicyController) deleteNamespace(old interface{}) {
 	}
 }
 
+// filterACNPsByServiceAccount gets all ClusterNetworkPolicies that will need to be
+// re-processed based on if they are affected by an added/updated/deleted
+// ServiceAccount.
+func (n *NetworkPolicyController) filterACNPsByServiceAccount(sa *v1.ServiceAccount) (acnps []*crdv1alpha1.ClusterNetworkPolicy) {
+	acnpObjs, err := n.cnpInformer.Informer().GetIndexer().ByIndex(HasServiceAccountsPeerIndex, strconv.FormatBool(true))
+	if err != nil {
+		klog.ErrorS(err, "Error fetching ClusterNetworkPolicies that have ServiceAccounts in either AppliedTo or Rules")
+		return nil
+	}
+
+	peerSelectedServiceAccount := func(peers []crdv1alpha1.NetworkPolicyPeer) bool {
+		for _, peer := range peers {
+			if peer.ServiceAccounts != nil {
+				if peer.ServiceAccounts.NamespacedName != nil && peer.ServiceAccounts.NamespacedName.Name == sa.Name && peer.ServiceAccounts.NamespacedName.Namespace == sa.Namespace {
+					return true
+				}
+				if peer.ServiceAccounts.NamespacedObjectSelector != nil {
```

Comment:
I am worried this is too much overhead for a blocking operation. Can we take advantage of the groupIndex to update the appliedToGroups/addressGroups instead of filtering by selector here?

Reply:
Haven't come up with a good idea. Anything on your side?